app.py

```diff
@@ -437,19 +468,26 @@ def generate_dependency_graph(selected_libraries):
     [dash.dependencies.State("library-select", "value")],
 )
 def update_multi_options(search_value, value):
+    all_values = None
     if not search_value:
         raise PreventUpdate
-    res = requests.get("https://libraries.io/api/search?q={lib_name}&sort=stars".format(lib_name=search_value)).json()
-    new_libraries = [r["name"] for r in res]
-
-    exact_search = requests.get('https://libraries.io/api/search?q={}'.format(search_value)).json()
-    if search_value in [r["name"] for r in exact_search]:
-        new_libraries.append(search_value)
-
-    all_libraries = list(set(value + new_libraries))
-    all_libraries.sort(key=len)
-    all_values = [{"label": r, "value": r} for r in all_libraries]
-    return all_values
+    try:
+        if len(search_value) < 3:
+            return [{"label": i, "value": i} for i in libraries]
+        req_url = "https://libraries.io/api/search?q={lib_name}&sort=stars".format(lib_name=search_value)
+        res = get_response(req_url).json()
+        new_libraries = [r["name"] for r in res]
+        search_url = 'https://libraries.io/api/search?q={}'.format(search_value)
```

Comment:
There's no API key here, which is the root cause of the timeout, I believe.

Reply:
added api key
and also reused method and using LRU cache